process/sync/baseSync.go

```diff
@@ -425,10 +420,10 @@ func (boot *baseBootstrap) ShouldSync() bool {
 }
 
 func (boot *baseBootstrap) removeHeaderFromPools(header data.HeaderHandler) []byte {
-	value, _ := boot.headersNonces.Get(header.GetNonce())
-	boot.headersNonces.Remove(header.GetNonce())
+	syncMap, _ := boot.headersNonces.Get(header.GetNonce())
```

Comment:
should treat if error.

Reply:
Refactored by bug/EN-2910-Fix-bug-when-nil-hash-is-return-from-sync-map #282